### PR TITLE
[MRG] Fix/surf plot thresholds

### PR DESCRIPTION
**Pull request by juhuntenburg**
This addresses issues mentioned in #1738 
(and https://neurostars.org/t/thresholded-surface-plot-with-nilearn/2992).

- Accidental thresholding of zeros removed

- Changed behaviour when no background map. Instead of a solid grey background (which removes 3D structure of the surface) the default is to display thresholded maps on grey half-transparent background. This is also shown in the example plot_surf_stat_map now. 

- Some additional small bugs and inconsistencies were removed

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
This needs an entry in the whats_new.

File: nilearn/plotting/surf_plotting.py [hunk 1]
```diff
@@ -343,7 +344,7 @@ def plot_surf_stat_map(surf_mesh, stat_map, bg_map=None,
     view: {'lateral', 'medial', 'dorsal', 'ventral', 'anterior', 'posterior'}, default is 'lateral'
         View of the surface that is rendered.
 
-    threshold : a number, None, or 'auto', default is None
+    threshold : a number or None, default is None
```
Comment (GaelVaroquaux) [inline nilearn/plotting/surf_plotting.py]: Is "auto" now an invalid value? In which case, it should be mentioned in the whats_new.

**Comment by GaelVaroquaux** [comment]
LGTM aside from the comments about the whats_new.

**Comment by KamalakerDadi** [comment]
Sorry for the documentation mess. It is fixed now in master. You have to rebase/merge master.

File: nilearn/plotting/surf_plotting.py (hunk 1 above)
Comment (juhuntenburg) [reply nilearn/plotting/surf_plotting.py]: As far as I can see it was never a valid input, just a fluke in the docstring

**Comment by juhuntenburg** [comment]
Yes, on it

**Comment by juhuntenburg** [comment]
The appveyor failure seems unrelated to this PR?

> ERROR: nilearn.datasets.tests.test_atlas.test_fetch_atlas_pauli_2017

**Comment by kchawla-pi** [comment]
Yeah, its not related to your PR.
I have restarted the build, we will see.

**Comment by kchawla-pi** [comment]
Its queued to run, don't wait for it.

**Comment by juhuntenburg** [comment]
Seems fine now, small error because I had another instance of the broken function reference in whats_new.rst, which I removed now

**Comment by GaelVaroquaux** [comment]
Gr8. Merging.

**Pull request merged**